ql/src/java/org/apache/hadoop/hive/ql/exec/tez/ColumnarSplitSizeEstimator.java

```diff
@@ -35,6 +35,9 @@ public class ColumnarSplitSizeEstimator implements SplitSizeEstimator {
   @Override
   public long getEstimatedSize(InputSplit inputSplit) throws IOException {
     long colProjSize = inputSplit.getLength();
+    if (colProjSize == 0) {
```

Comment:
The original code had two paths to overwrite the value provided from `inputSplit.getLength` and now we skip those paths. Also, what if the columnar projection size or the inner split has 0 bytes? And the original root cause of the issue is at the end of this method:
```java
    if (colProjSize <= 0) {
      /* columnar splits of unknown size - estimate worst-case */
      return Integer.MAX_VALUE;
    }
```

What about changing `colProjSize <= 0` to `colProjSize < 0` so that we can keep the original logic and fix the Integer.MAX_VALUE issue?

Reply:
Hi，thanks for review the code

The reason for judging the split length is 0 at the beginning is to keep the original logic.

I think，the original logic is that if the columnar projection size or the inner split has 0 bytes, it does not mean that the length of this split is 0. Returning Integer.MAX_VALUE is a safer method.

Reply:
Thank you, I didn't know about that behaviour.

Reply:
I think that 0 things was explicitly added as part of https://issues.apache.org/jira/browse/HIVE-13821 for ACID